aip/client-libraries/4222.md

```diff
@@ -154,6 +160,46 @@ If none of the routing parameters matched their respective fields, the routing h
 Much like URL parameters, if there is more than one key-value pair to be sent, the `&`
 character is used as the separator.
 
+### `path_template` syntax
+
+As seen in the above examples, the `path_template` can use a variety of symbols
+that are interpreted by code generators during conversion to regular expressions
+or non-regular expression matcher implementations. The syntax for these symbols
+is as follows:
+
+- The only acceptable segment delimiter is `/`.
+  - The last symbol in a `path_template` **may** be a delimiter - it will be 
+    ignored.
+  - A segment **must not** represent a complex resource ID as described in
+    [AIP-4231][]. A Generator **should** emit an error in this case.
+- A segment **must** be of one of the following types:
+  - `*`: A single-segment wildcard. Corresponds to 1 or more non-`/` symbols.
+    The regex describing it is `[^\/]+`.
+    - A Single-segment wildcard typically represents a resource ID.
+  - `**`: A multi-segment wildcard. Corresponds to 0 or more symbols or segments
+    (so it can be empty). Consumes segment delimiters and does not match unless
+    the delimiter has trailing content. The regular expression is `([:\/].*)?`.
+    - A multi-segment wildcard **must** only appear as the final segment.
+  - `LITERAL`: A literal string. A literal can consist of any alphanumeric
+    symbol.
+    - A literal **must not** contain a symbol reserved in this syntax.
+    - Literal segments typically represent a resource collection ID or base
+      path.
+  - `{}`: A variable. This matches part of the path as specified by its
```

Comment:
I'd describe it as a "variable segment" and not a "variable", here and everywhere below.

Reply:
For sure, done.